### PR TITLE
fix(native-validation): clear error

### PR DESCRIPTION
**Pull request by jorisre**
Related to https://github.com/react-hook-form/resolvers/issues/182

### CODE REVIEW

File: src/__tests__/validateFieldsNatively.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+};
+
+const fields = {
+  name: {
+    ref: {
+      reportValidity: jest.fn(),
+      setCustomValidity: jest.fn(),
+    },
+  },
+  nd: {
```
Comment (bluebill1049) [inline src/__tests__/validateFieldsNatively.ts:15]: minor stuff: `nd` means? 😆

File: src/validateFieldsNatively.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    const field = options.fields[fieldPath];
+
+    if (field && field.ref && 'reportValidity' in field.ref) {
+      const error = get(errors, fieldPath) as FieldError | undefined;
+
+      if (error) {
+        field.ref.setCustomValidity(error.message || '');
+      } else {
+        field.ref.setCustomValidity('');
+      }
```
Comment (bluebill1049) [inline src/validateFieldsNatively.ts:18]:
```suggestion
      field.ref.setCustomValidity((get(errors, fieldPath) as FieldError | undefined) ? error.message || '' : '');
```

save few bytes :)

**Review by bluebill1049 — APPROVED** [review]
Awesome stuff! thanks a lot for the PR.

File: src/__tests__/validateFieldsNatively.ts (hunk 1 above)
Comment (jorisre) [reply src/__tests__/validateFieldsNatively.ts:15]: It's just for testing purpose, I'll improve it :)

Comment (bluebill1049) [reply src/__tests__/validateFieldsNatively.ts:15]: hahah don't worry about it.

File: src/validateFieldsNatively.ts (hunk 2 above)
Comment (jorisre) [reply src/validateFieldsNatively.ts]: Done ✅

**Review by bluebill1049 — APPROVED** [review]
🚀

**Pull request merged**